clash-cores/src/Clash/Cores/Xilinx/Ila.hs

```diff
@@ -0,0 +1,141 @@
+{-|
+Copyright  :  (C) 2023, Google Inc,
+License    :  BSD2 (see the file LICENSE)
+Maintainer :  QBayLogic B.V. <devops@qbaylogic.com>
+
+Support for [Xilinx Integrated Logic Analyzer v6.2](https://docs.xilinx.com/v/u/en-US/pg172-ila).
+An Integrated Logic Analyzer (ILA) is a feature provided by Xilinx in its design
+tools, notably Vivado, that allows designers to debug their FPGA logic in
+real-time. It stores the signals it samples in a ring buffer, allowing users to
+see values before and after a trigger point.
+
+It is necessary to read the product guide linked above in order to effectively
+use the IP. Clash simulation is not applicable for this IP.
+
+When using the generated ILAs make sure you have set the correct JTAG clock speed:
+[/"For non-Versal architectures, if your design contains debug cores, ensure that the JTAG clock is 2.5 times slower than the debug hub clock."/](https://www.xilinx.com/content/dam/xilinx/support/documents/sw_manuals/xilinx2022_2/ug908-vivado-programming-debugging.pdf)
+
+-}
+
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE DeriveLift #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE ViewPatterns#-}
+
+-- See [Note: eta port names for trueDualPortBlockRam]
+{-# OPTIONS_GHC -fno-do-lambda-eta-expansion #-}
+
+module Clash.Cores.Xilinx.Ila
+  ( ila
+
+  -- * Config
+  , IlaConfig(..)
+  , ilaConfig
+  , ProbeType(..)
+  , Depth(..)
+
+  -- * Utilities
+  , Ila(..)
+  ) where
+
+import Clash.Explicit.Prelude
+
+import Clash.Annotations.Primitive (Primitive (InlineYamlPrimitive))
+
+import Data.String.Interpolate (__i)
+
+import Clash.Cores.Xilinx.Ila.Internal
+
+-- | A default ILA config that:
+--
+--  * Configures no pipeline registers
+--  * Stores 4096 samples
+--  * Sets 2 comparators per probe
+--  * Sets all probes to be suited for DATA and TRIGGER
+--  * Enables capture control
+--
+-- See 'IlaConfig' for more information.
+ilaConfig :: Vec n String -> IlaConfig n
+ilaConfig names = IlaConfig
```

Comment:
I find it strange to have a default configuration, where one of the options still must be provided as an argument. It just makes the usage kind of cumbersome. I would either perfer having some predefined names or sometthing like 
```haskell
ilaConfig = IlaConfig {
  ...
  , probeNames = error "Missing probe names"
  ... 
}

Reply:
Just forgot: It might also be reasonable to change the module name to `Clash.Cores.Xilinx.ILA` to make it consistent with `Clash.Cores.Xilinx.VIO`. Also because a big I and a small l after each other being a hard to read for many font settings :grin:.

Reply:
Right, we've settled on the other way around :sweat_smile:, because all caps can make it hard to see where the "borders" between concepts lie. I.e., `ILATCL` vs `IlaTcl`. In a future PR we're going to make everything consistent in `clash-cores`.

Reply:
Concerning the config: I don't think we should introduce partial functions / structures. You _have_ to supply it either way - through a record update or by function argument. The latter will give you a type error if you mess up, the first won't.